src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,183 @@
+package duke;
+
+import duke.exceptions.DukeException;
+
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
+
+import duke.execution.commands.Command;
+import duke.execution.Parser;
+import duke.execution.Storage;
+import duke.execution.TaskList;
+import duke.execution.Ui;
+
+import java.io.IOException;
+
+/**
+ * Main class to handle other classes.
+ */
+public class Duke extends Application {
+
+    static final String TEXT_DOCUMENT = "./src/main/java/data/data.txt";
+
+    /** Storage to read and write files. */
+    private Storage storage;
+    /** Collections of the tasks. */
+    private TaskList tasks;
+    /** Handles input and output. */
+    private Ui ui;
+    /** Action taken depending on input. */
+    private Command command;
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+    /**
+     * Constructor for JavaFX implementation
+     */
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage(TEXT_DOCUMENT);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException | IOException e) {
+            ui.showLoadingError(e);
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Initialising of programme.
+     * @param filePath Filepath of data stored.
+     */
+    public Duke(String filePath) {
```

Comment:
The logic of both constructors are similar. As a suggestion, you can consider using just one constructor to process the data.